src/components/simplesearchengine/SimpleSearchEngineSecondary.java

```diff
@@ -0,0 +1,15 @@
+package components.simplesearchengine;
+
+/**
+ * {@code SimpleSearchEngineKernel<T>} enhanced with secondary methods.
+ *
+ * @param <T>
+ *                type of {@code SimpleSearchEngine} entries
+ *
+ * @author Saurav Poudyel
+ *
+ */
+public abstract class SimpleSearchEngineSecondary<T>
+                implements SimpleSearchEngine<T> {
+
```

Comment:
You should implement the common methods here--at least `.equals()` and `toString()`. I can see the argument for not implementing `hashCode()`.